File: marketing-backend/src/main/java/edu/yavirac/oportunidadbackend/feature/productos/Productos.java
package edu.yavirac.oportunidadbackend.feature.productos;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.MappedCollection;
import org.springframework.data.relational.core.mapping.Table;
import lombok.Data;

@Data
@Table("productos")
public class Productos{

    @Id
    @Column("producto_id")
    private long productoId;

    private String nombre;
    private String descripcion;
    private String marca;
    private String modelo;

    private Date created;
    private Date updated;
    private boolean enable;
    @Column("persons_id")
    private Long creadoPor;


    @MappedCollection(idColumn = "producto_id")
    private Set<ProductosRelacionado> relacionados = new HashSet<>();


}

File: marketing-frontend/src/app/feature/productos/productos.ts
import { ProductosRelacionados } from "./ProductosRelacionados";

export interface Productos {
  productoId: number,
  nombre: string,
  descripcion: string,
  marca: string,
  modelo: string,
  created: Date,
  updated: Date,
  enable: boolean,
  personsId: number,
  relacionados: ProductosRelacionados[]



}
